File: app/src/main/java/piuk/blockchain/android/data/websocket/models/EthWebsocketResponse.java
package piuk.blockchain.android.data.websocket.models;

import com.fasterxml.jackson.annotation.JsonAutoDetect;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.math.BigInteger;

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties(ignoreUnknown = true)
@JsonAutoDetect(
        fieldVisibility = JsonAutoDetect.Visibility.NONE,
        getterVisibility = JsonAutoDetect.Visibility.NONE,
        setterVisibility = JsonAutoDetect.Visibility.NONE,
        creatorVisibility = JsonAutoDetect.Visibility.NONE,
        isGetterVisibility = JsonAutoDetect.Visibility.NONE
)
public class EthWebsocketResponse {

    @JsonProperty("op")
    private String op;
    @JsonProperty("account")
    private String account;
    @JsonProperty("balance")
    private BigInteger balance;
    @JsonProperty("nonce")
    private Long nonce;
    @JsonProperty("txHash")
    private String txHash;
    @JsonProperty("tx")
    private EthWebsocketTx tx;

    public String getOp() {
        return op;
    }

    public String getAccount() {
        return account;
    }

    public BigInteger getBalance() {
        return balance;
    }

    public Long getNonce() {
        return nonce;
    }

    public String getTxHash() {
        return txHash;
    }

    public EthWebsocketTx getTx() {
        return tx;
    }

}

File: app/src/main/java/piuk/blockchain/android/data/websocket/models/EthWebsocketResponse.kt
package piuk.blockchain.android.data.websocket.models

import com.fasterxml.jackson.annotation.JsonAutoDetect
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonInclude

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties(ignoreUnknown = true)
@JsonAutoDetect(
        fieldVisibility = JsonAutoDetect.Visibility.NONE,
        getterVisibility = JsonAutoDetect.Visibility.NONE,
        setterVisibility = JsonAutoDetect.Visibility.NONE,
        creatorVisibility = JsonAutoDetect.Visibility.NONE,
        isGetterVisibility = JsonAutoDetect.Visibility.NONE
)
data class EthWebsocketResponse(
        val op: String,
        val account: String,
        val balance: String,
        val nonce: Long,
        val txHash: String,
        val tx: EthWebsocketTx
)

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties(ignoreUnknown = true)
@JsonAutoDetect(
        fieldVisibility = JsonAutoDetect.Visibility.NONE,
        getterVisibility = JsonAutoDetect.Visibility.NONE,
        setterVisibility = JsonAutoDetect.Visibility.NONE,
        creatorVisibility = JsonAutoDetect.Visibility.NONE,
        isGetterVisibility = JsonAutoDetect.Visibility.NONE
)
data class EthWebsocketTx(
        val blockHash: String,
        val blockNumber: Long,
        val from: String,
        val gas: Long,
        val gasPrice: String,
        val hash: String,
        val input: String,
        val nonce: Long,
        val to: String,
        val transactionIndex: Long,
        val value: String,
        val v: String,
        val r: String,
        val s: String
)